tidy3d/plugins/mode/solver.py

```diff
@@ -291,6 +302,21 @@ def solver_diagonal(
     ):  # pylint:disable=too-many-arguments
         """EM eigenmode solver assuming ``eps`` and ``mu`` are diagonal everywhere."""
 
+        # code associated with these options is included below in case it's useful in the future
+        enable_incidence_matrices = False
+        enable_preconditioner = False
+        analyze_conditioning = False
+
+        def incidence_matrix_for_pec(eps_vec, threshold=0.9 * pec_val):
+            """Incidence matrix indicating non-PEC entries associated with 'eps_vec'."""
+            nnz = eps_vec[eps_vec > threshold]
```

Comment:
It should now be `[np.abs(eps_vec) > threshold]` as you may have made the epsilon conductive by now?

Reply:
Good catch, thanks